### PR TITLE
[REF-2141] Custom component command improvements

### PR DESCRIPTION
**Pull request by martinxu9**
## Summary
- If enclosing folder already starts with reflex, do not add that prefix into the name parts. corner case, if they name their folder reflex, we can't derive a package name for them, just prompt user to add --library-name
- Add a flag to publish command --build/--no-build, defaults to `build==True`. When publishing, check if build already exists (has to be the same version), if already exists, ask user if they want to build anyway (default is "n")
- add package-name to the requirements.txt in the demo app
- Add .gitignore in the top directory to include the dist folder and some other generated files during build
- polish the summary message for init to make it more clear the next steps. Some of the information needs to be manually updated.
## Tests
- Tested manually
- The console print now looks like this
```
~/test/ $ reflex component init
────────────────────────────────────────── Initializing reflex-test project ───────────────────────────────────────────
Info: Populating pyproject.toml with package name: reflex-test
Info: Initializing the component directory: custom_components/reflex_test
Info: Creating app for testing: test_demo
─────────────────────────────────────────────── Initializing test_demo ────────────────────────────────────────────────
Warning: Your version (0.4.2) of reflex is out of date. Upgrade to 0.4.3 with 'pip install reflex --upgrade'
[23:50:29] Initializing the app directory.                                                                console.py:85
           Initializing the web directory.                                                                console.py:85
Success: Initialized test_demo
────────────────────────────────────── Installing reflex-test in editable mode. ───────────────────────────────────────
Info: Package reflex-test installed!
Custom component initialized successfully!
─────────────────────────────────────────────────── Project Summary ───────────────────────────────────────────────────
[ README.md ]: Package description. Please add usage examples.
[ pyproject.toml ]: Project configuration file. Please fill in details such as your name, email, homepage URL.
[ custom_components/ ]: Custom component code template. Start by editing it with your component implementation.
[ test_demo/ ]: Demo App. Add more code to this app and test.
```

### CODE REVIEW

**Review by masenf — APPROVED** [review]

File: reflex/custom_components/custom_components.py [hunk 1]
```diff
@@ -456,14 +483,72 @@ def _validate_credentials(
...
+        Exit: If the version is not found in the pyproject.toml.
+
+    Returns:
+        The version to publish.
+    """
+    # Get the version from the pyproject.toml.
+    version_to_publish = None
+    with open(CustomComponents.PYPROJECT_TOML, "r") as f:
+        pyproject_toml = f.read()
+        match = re.search(r'version = "(.*?)"', pyproject_toml)
```
[the remark this answers is not in the document]
Comment (martinxu9) [reply reflex/custom_components/custom_components.py:499]: oh thanks, copilot wrote it and I just kinda took it. I think quotes are important. I have certain plugs that seem to clean up quotes in non python files differently from time to time. It'll be silly to break over that. 
I initially wanted to use the toml package, but felt really hesitant I'll be asked to jump through hoops to get all of this into a separate package again just to reduce the payload. Didn't know it's only from 3.11, then we'll think of something else. I think full parsing of the toml file is good, then before publishing, I can verify the user information for them, only on pypi, there's no way to change the name, email, URL stuff anymore. I feel that could be part of the workflow. 

Thanks for the review. I'll fix the regex before merging. I feel it's serious enough bug now that I know it.

**Comment by martinxu9** [comment]
@masenf revision uploaded.

**Review by masenf — APPROVED** [review]

**Review by Alek99 — CHANGES_REQUESTED** [review]
In the src.py.jinja2 we import only `import reflex as rx` I think we need to add a few of the common reflex imports that are used in almost every component such as  `from reflex.vars import Var`

This is all I can think of now we need to add

**Comment by martinxu9** [comment]
> In the src.py.jinja2 we import only `import reflex as rx` I think we need to add a few of the common reflex imports that are used in almost every component such as `from reflex.vars import Var` I see we have the example Var commented out but if they un comment it would be missing the import
> 
> This is all I can think of now we need to add

The template code in the comment already has rx.Var as examples, dont think we need to add just the Var imports here

**Review by Alek99 — APPROVED** [review]
Awesome!

**Comment by Alek99** [comment]
> > In the src.py.jinja2 we import only `import reflex as rx` I think we need to add a few of the common reflex imports that are used in almost every component such as `from reflex.vars import Var` I see we have the example Var commented out but if they un comment it would be missing the import
> > This is all I can think of now we need to add
> 
> The template code in the comment already has rx.Var as examples, dont think we need to add just the Var imports here

Yeah your right missed totally missed that

**Pull request merged**